client/components/plans/plan-actions/index.jsx

```diff
@@ -188,10 +209,12 @@ module.exports = React.createClass( {
 	},
 
 	freePlanExpiration: function() {
-		if ( ! this.planHasCost() ) {
-			return (
-				<span className="plan-actions__plan-expiration">{ this.translate( 'Never expires', { context: 'Expiration info for free plan in /plans/' } ) }</span>
-			);
+		if ( config.isEnabled( 'upgrades/free-trials' ) ) {
+			if ( ! this.planHasCost() ) {
```

Comment:
I thought it's because when the trials are not enabled, the other plans don't have any text about their subscription length or expiry under them so it would look out of place to have it for the free trial. 

And the same thing the other way around when the trials are enabled, the Never expires text is there just so the free plan description looks like the other two, ie. is of the same length but doesn't just have empty white space under it.